### PR TITLE
Refactor config

### PR DESCRIPTION
**Pull request by wangmq0719**

### CODE REVIEW

File: qingstor/sdk/config.py [hunk 1]
```diff
@@ -96,8 +105,15 @@ def load_default_config(self):
         self.load_config_from_data(config_data)
         return self
 
+    def load_from_env(self, key):
+        key = f"QINGSTOR_{key.upper()}"
+        v = os.environ.get(key)
+        return v
```
Comment (Xuanwo) [inline qingstor/sdk/config.py:111]:
```suggestion
        return os.environ.get(key)
```

File: qingstor/sdk/config.py [hunk 2]
```diff
@@ -87,7 +89,14 @@ def install_default_user_config(self):
...
     def load_config_from_data(self, data):
         for (k, v) in data.items():
+            value = self.load_from_env(k)
+            if not value is None:
+                v = value
             setattr(self, k, v)
+        if self.endpoint != "":
+            self.protocol = urlparse(self.endpoint).scheme
+            self.host = urlparse(self.endpoint).hostname
+            self.port = urlparse(self.endpoint).port
```
Comment (Xuanwo) [inline qingstor/sdk/config.py:99]:
```suggestion
            u = urlparse(self.endpoint)
            (self.protocol, self.host, self.port)  = (u.scheme, u.hostname, u.port)
```

File: qingstor/sdk/config.py (hunk 1 above)
Comment (Xuanwo) [reply qingstor/sdk/config.py:111]: BTW, we should not load `host`, `port` or `protocol` from env as them will be remove in later releases.

File: qingstor/sdk/config.py [hunk 3]
```diff
@@ -87,7 +89,14 @@ def install_default_user_config(self):
 
     def load_config_from_data(self, data):
         for (k, v) in data.items():
+            value = self.load_from_env(k)
+            if not value is None:
```
Comment (Xuanwo) [inline qingstor/sdk/config.py:93]:
```suggestion
            if value is not None:
```

Comment (Xuanwo) [reply qingstor/sdk/config.py:93]: BTW, what will happen if user's config do not have key `endpoint`?

Comment (wangmq0719) [reply qingstor/sdk/config.py:93]: if user's config have endpoint, we use endpoint to replace protocol, host and port. if user's config do not have key endpoint,We use protocol, host and port config directly

File: qingstor/sdk/config.py (hunk 1 above)
Comment (wangmq0719) [reply qingstor/sdk/config.py:111]: I think it's better not to remove them, but to keep forward compatibility

File: qingstor/sdk/config.py (hunk 3 above)
Comment (Xuanwo) [reply qingstor/sdk/config.py:93]: But `data` here is read from user's config, if user's config don't have `endpoint`, we will not read this value from env.

Think about the case like following:

user has config:

```yaml
access_key_id: abc
secret_access_key: def
host: qingstor.com
```

The user start program with env `endpoint: abc.com`, current implemation could ignore `endpoint` and use `host` directly.

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**